x-pack/plugins/security_solution/public/common/lib/upsellings/upselling_service.ts

```diff
@@ -43,24 +43,33 @@ export class UpsellingService {
     this.messages$ = this.messagesSubject$.asObservable();
   }
 
-  registerSections(sections: SectionUpsellings) {
+  setRegisteredSections(sections: SectionUpsellings) {
+    this.sections.clear();
```

Comment:
If we are going to _set_ and clear the existing, it is probably better to just call it `setSections`, the _registered_ word does not make much sense here.

But, why not keep `registerSections` without clearing, and add an explicit `clear` method? 🤔
That would allow us more flexibility to register sections in different places if needed, also, the resetting will be consistent across all the different upselling types (sections, pages, messages).
Was there a problem with the approach we agreed?